recipes/llama_api_providers/examples_with_aws/Prompt_Engineering_with_Llama_3_On_Amazon_Bedrock.ipynb

```diff
@@ -0,0 +1,2358 @@
+    "response_input_guard = llm.predict(payload_input_guard)\n",
+    "llama_guard_response = response_input_guard[0][\"generated_text\"].split(\"[/INST]\",1)[1]\n",
+    "print(llama_guard_response)\n",
+    "print('\\n\\n')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 320,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "==============\n",
+      "I forgot how to kill a process in Linux, can you help in a few short steps?\n",
+      "==============\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Llama Guard has determined the input prompt is safe, so let's now send it to Bedrock\n",
+    "if llama_guard_response == 'safe':\n",
+    "    prompt = messages_input[0]['content']\n",
+    "    bedrock_output = complete_and_print(prompt, verbose=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 326,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Bedrock Generation Safety Response from Llama Guard: safe\n",
+      "\n",
+      "\n",
+      "]\n",
+      "\n",
+      "Here are the steps to kill a process in Linux:\n",
+      "\n",
+      "**Method 1: Using the `kill` command**\n",
+      "\n",
+      "1. Open a terminal and type `ps -ef | grep <process_name>` to find the process ID (PID) of the process you want to kill. Replace `<process_name>` with the actual name of the process.\n",
+      "2. Once you have the PID, type `kill <PID>` to send a signal to the process to terminate.\n",
+      "\n",
+      "**Method 2: Using the `pkill` command**\n",
+      "\n",
+      "1. Open a terminal and type `pkill <process_name>` to find and kill the process by its name.\n",
+      "2. If there are multiple processes with the same name, you can specify the PID by adding the `-pid` option, like this: `pkill -pid <PID> <process_name>`.\n",
+      "\n",
+      "**Method 3: Using the `killall` command**\n",
+      "\n",
+      "1. Open a terminal and type `killall <process_name>` to kill all processes with the same name.\n",
+      "2. Be careful with this method, as it can terminate multiple processes with the same name.\n",
+      "\n",
+      "Remember to replace `<process_name>` with the actual name of the process you want to kill.\n",
+      "\n",
+      "If the process is not responding to the `kill` signal, you can try using the `-9` option to send a `SIGKILL` signal, which is more forceful: `kill -9 <PID>`.\n",
+      "\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Now before we decide to return the response from the inference, we can run the\n",
+    "# generated output through Llama Guard again\n",
+    "\n",
+    "messages_input = [{\"role\": \"user\", \"content\": bedrock_output}]\n",
+    "payload_input_guard = {\"inputs\": format_guard_messages(messages_input)}\n",
```

Comment:
Ideally, the user prompt should be in this list as well, to showcase that when evaluating the model output, the user input is also taken into consideration.